metals/src/main/scala/scala/meta/internal/rename/RenameProvider.scala

```diff
@@ -77,105 +78,121 @@ final class RenameProvider(
       params: RenameParams,
       token: CancelToken
   ): Future[WorkspaceEdit] = {
-    if (!compilations.currentlyCompiling.isEmpty) {
-      client.showMessage(isCompiling)
-      Future.successful(new WorkspaceEdit())
-    } else {
-      val source = params.getTextDocument.getUri.toAbsolutePath
-      definitionProvider.definition(source, params, token).map { definition =>
-        val textParams = new TextDocumentPositionParams(
-          params.getTextDocument(),
-          params.getPosition()
-        )
-
-        val symbolOccurrence =
-          definitionProvider.symbolOccurrence(source, textParams.getPosition)
-
-        val suggestedName = params.getNewName()
-        val newName =
-          if (suggestedName.charAt(0) == '`')
-            suggestedName.substring(1, suggestedName.length() - 1)
-          else suggestedName
-
-        def includeSynthetic(syn: Synthetic) = {
-          syn.tree match {
-            case SelectTree(_, id) =>
-              id.exists(_.symbol.desc.name.toString == "apply")
-            case _ => false
+    def compilationFinishedSuccessfully(
+        source: AbsolutePath
+    ): Future[Boolean] = {
+      if (compilations.currentlyCompiling.isEmpty) {
+        Future(true)
+      } else {
+        compilations.cascadeCompileFiles(Seq(source)).map { compilationResult =>
+          if (compilationResult.getStatusCode() == StatusCode.OK) {
```

Comment:
We most likely don't have to fail in case of compile errors. SemanticDB files will still be available and rename will only fail in case of new non-compiled symbols, which is the current behavior.

Reply:
removed check of compilation result